### PR TITLE
add assume_role_arn and external_id docs reference

### PR DESCRIPTION
**Pull request by GavinFrazar**
This is a docs only PR that adds `assume_role_arn` and `external_id` in database service matchers, static config, and dynamic database spec.

Issue: https://github.com/gravitational/teleport/issues/21872

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: docs/pages/database-access/reference/configuration.mdx [hunk 1]
```diff
@@ -135,6 +135,12 @@ spec:
   aws:
     # Region the database is deployed in.
     region: "us-east-1"
+    # Optional AWS role that the database service will assume to access
```
Comment (ptgott) [inline docs/pages/database-access/reference/configuration.mdx:138]:
```suggestion
    # Optional AWS role that the Database Service will assume to access
```

File: docs/pages/database-access/reference/configuration.mdx [hunk 2]
```diff
@@ -135,6 +135,12 @@ spec:
...
+    # this database.
+    assume_role_arn: "arn:aws:iam::123456789012:role/example-role-name"
+    # Optional AWS external ID that the database service will use to assume
+    # a role in an external AWS account.
```
Comment (ptgott) [inline docs/pages/database-access/reference/configuration.mdx:142]:
```suggestion
    # Optional AWS external ID that the Database Service will use to assume
    # a role in an external AWS account.
```

File: docs/pages/includes/config-reference/database-config.yaml [hunk 3]
```diff
@@ -26,6 +26,14 @@ db_service:
   - types: ["rds", "rdsproxy","redshift", "redshift-serverless", "elasticache", "memorydb"]
     # AWS regions to register databases from.
     regions: ["us-west-1", "us-east-2"]
+    # Optional AWS role that the database service will assume to discover
+    # and register AWS-hosted databases.
+    # Discovered databases are also accessed by the database service via
+    # this role.
```
Comment (ptgott) [inline docs/pages/includes/config-reference/database-config.yaml:32]:
```suggestion
    # Optional AWS role that the Database Service will assume to discover
    # and register AWS-hosted databases.
    # Discovered databases are also accessed by the Database Service via
    # this role.
```

File: docs/pages/includes/config-reference/database-config.yaml [hunk 4]
```diff
@@ -26,6 +26,14 @@ db_service:
...
+    assume_role_arn: "arn:aws:iam::123456789012:role/example-role-name"
+    # Optional AWS external ID that the database service will use to assume
```
Comment (ptgott) [inline docs/pages/includes/config-reference/database-config.yaml:34]:
```suggestion
    # Optional AWS external ID that the Database Service will use to assume
```

File: docs/pages/includes/config-reference/database-config.yaml [hunk 5]
```diff
@@ -92,6 +100,12 @@ db_service:
     aws:
       # Region the database is deployed in.
       region: "us-east-1"
+      # Optional AWS role that the database service will assume to access
+      # this database.
+      assume_role_arn: "arn:aws:iam::123456789012:role/example-role-name"
+      # Optional AWS external ID that the database service will use to assume
+      # a role in an external AWS account.
+      external_id: "example-external-id"
```
Comment (ptgott) [inline docs/pages/includes/config-reference/database-config.yaml:108]:
```suggestion
      # Optional AWS role that the Database Service will assume to access
      # this database.
      assume_role_arn: "arn:aws:iam::123456789012:role/example-role-name"
      # Optional AWS external ID that the Database Service will use to assume
      # a role in an external AWS account.
      external_id: "example-external-id"
```

**Review by ptgott — APPROVED** [review]
Approved with suggestions to replace "database service" with "Database Service" so product names are consistent across the docs

**Pull request merged**